swig/cpp/src/Session.cpp

```diff
@@ -427,12 +427,14 @@ shared_ptr<Iter_Change> Subscribe::get_changes_iter(const char *xpath)
     }
 }
 
-shared_ptr<Operation> Subscribe::get_change_next(shared_ptr<Iter_Change> iter, shared_ptr<Val_Holder> new_value,\
-                                            shared_ptr<Val_Holder> old_value)
+shared_ptr<Operation> Subscribe::get_change_next(shared_ptr<Iter_Change> iter, shared_ptr<Val> new_value,\
```

Comment:
I was thinking about this and I believe that the cleanest solution would be to slightly modify the Operation class (renaming it to Change might also be suitable). Basically, I was thinking about adding  shared_ptr<Val> new_val and old_val into the Change class. 

In this case all output would be returned using return value and user wouldn't need to provide "output-like arguments".

```C++
shared_ptr<Change> Subscribe::get_change_next(shared_ptr<Iter_Change> iter)
````
Would do you think? Would it cause any issues?